python/ray/workflow/workflow_access.py

```diff
@@ -121,14 +123,40 @@ class LatestWorkflowOutput:
     step_id: "StepID"
 
 
+def _get_workflow_storage(workflow_id: str) -> workflow_storage.WorkflowStorage:
+    # Avoid initialize workflow in the workflow management actor,
+    # because it would invoke remote methods of itself,
+    # resulting in deadlock.
+    return workflow_storage.WorkflowStorage(workflow_id, _no_init_workflow=True)
+
+
+# TODO(suquark): what should be the default concurrency limit?
+@ray.remote(num_cpus=0)
+class WorkflowResultRespondingActor:
+    """This async actor is for responding pending workflow results."""
+
+    def __init__(self):
+        self.running_or_pending_workflows: Dict[str, asyncio.Future] = {}
+
+    def set_running_workflow_output(self, workflow_id: str, output: WorkflowStaticRef):
+        if workflow_id not in self.running_or_pending_workflows:
+            self.running_or_pending_workflows[workflow_id] = asyncio.Future()
+        self.running_or_pending_workflows[workflow_id].set_result(output)
+
+    def set_pending_workflow(self, workflow_id: str):
+        self.running_or_pending_workflows[workflow_id] = asyncio.Future()
```

Comment:
Can we make this async and then get_output seems not necessary

```
async  def set_pending_workflow(self, workflow_id: str):
       self.running_or_pending_workflows[workflow_id] = asyncio.Future()
       return await self.running_or_pending_workflows[workflow_id]
```

Reply:
hmmm, get output is used in another place for the `get_output` API. I still prefer to keep them seperated.

Reply:
this comment is outdated